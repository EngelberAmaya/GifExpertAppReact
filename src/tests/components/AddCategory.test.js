import AddCategory from '../../components/AddCategory';
import {shallow} from 'enzyme';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
      
    }); 
    
    // test('Debe cambiar la caja de texto', () => {

    //     const input =  wrapper.find('input');
    //     const value = 'Hola Mundo';

    //     input.simulate('change', {
    //         target: { 
    //             value: value
    //         }
    //     });

    //     expect(wrapper.find('p').text().trim()).toBe(value);
      
    // });

    test('NO debe de postear la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
      
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {

        const value = 'Hola Mundo';

        // simular el inputChange
        wrapper.find('input').simulate('change', { target: { value } });

        //simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        //setCategories se debe haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);

        //el valor del input debe estar ''
        expect(wrapper.find('input').prop('value')).toBe('');
      
    })
    
    
  
})
