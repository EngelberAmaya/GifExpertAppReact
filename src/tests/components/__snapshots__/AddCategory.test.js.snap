// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Escriba lo que desea buscar y luego presione enter"
    type="text"
    value=""
  />
</form>
`;
